src/routes/favorites.js

```diff
@@ -1,72 +1,70 @@
 import React from "react";
-import { useHistory } from 'react-router-dom';
+import {
+  Link,
+  useHistory
+} from 'react-router-dom';
 import {
   StudySpaceCard,
   Topbar
 } from "../components";
 
-// TODO: Read the JSON file from App.js
-var study_spaces = require("../study-spaces.json");
-var dataHtml = [];
+function handleClick() {
+  console.log("Favorites card clicked.");
+
+  // TODO: Commented this out because it was breaking the back button.
+  // history.push("/view-space");
+}
+
+/**
+ * Improvements:
+ *  - update list as soon as it's clicked instead of having to refresh the page
+ */
+function handleClickDelete(event) {
+  console.log("Favorites card deleted.");
 
+  // Prevents the Card's onClick/Link from being triggered
+  event.preventDefault();
+
+  // Delete from Local Storage
+  var favoriteSpacesJson = JSON.parse(localStorage.getItem('favoriteSpaces'));
+  var indexToDelete = event.target.getAttribute("index");
+  favoriteSpacesJson.data.splice(indexToDelete, 1);
+  localStorage.setItem('favoriteSpaces', JSON.stringify(favoriteSpacesJson))
+}
+/* Sources of Help:
+ *  - Interrupting the Link on the Card: https://stackoverflow.com/questions/39849108/disable-react-router-link-in-react
+ **/
+
+function favoritesListToHtml(data) {
+  return data.map((space, i) => {
+    return <Link to={{ state: { data : space }, pathname: "/view-space" }} style={{ textDecoration: "none", color: "#000000" }}>
+      <StudySpaceCard
+        index={i}
+        title={space.title}
+        description={space.description}
+        imageFilePath={space.imageFilePath}
+        distance={space.distance}
+        tags={space.tags}
+        handleClick={handleClick}
+        hasRemove={true}
+        handleClickDelete={handleClickDelete}
+      />
+    </Link>;
+  });
+}
 /*
  * Sources of Help:
  *  - Dynamically generating the list of study space cards: https://www.skptricks.com/2018/06/append-or-prepend-html-using-reactjs.html
  **/
 
-const Favorites = () => {
+const Favorites = (props) => {
+  // TODO: Figure out how to use this correctly with the handleClick().
```

Comment:
? in what way particularly do you need help figuring out

Reply:
What is the purpose of this history object?

Reply:
to go to the next page without completely reloading the app

Reply:
oh ok i see I'll add that back in then